shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -45,6 +45,75 @@
 
 }  // namespace
 
+/// A proxy class for SemanticsObject and UISwitch.  For most Accessibility and
+/// SemanticsObject methods it delegates to the semantics object, otherwise it
+/// sends messages to the UISwitch.
+@interface FlutterSwitchSemanticsObject : UISwitch
```

Comment:
I definitely lack some context here which is making it hard for me to review this. Why is this object a switch subclass? It is not going to be ever used as a switch. That is, it is not in the view hierarchy and you have already overridden most of ally methods. If you wanted to delegate to the base classes implementation of the a11y methods, the responses will be based on a switch in its default state (and not how it appears in the Flutter view hierarchy). Towards that end, shouldn't it be on the implementation to override all the a11y methods as necessary here?

Again. I could be missing something but the base class and the forwarding I don't get the point of.

Reply:
No problem, I can explain.  There is magic that happens on iOS for UISwitch with respect to VoiceOver.  If you look at the results of a UISwitch's accessibility methods, it doesn't match up to what VoiceOver actually says.  In order to get the magic the SemanticsObject has to be a UISwitch (it has undocumented methods or there is some reflection happening).  We do something similar for text fields.